client/src/app/pages/applications/components/application-analysis-status.tsx

```diff
@@ -34,3 +34,30 @@ export const ApplicationAnalysisStatus: React.FC<
 > = ({ state }) => {
   return <IconedStatus preset={getTaskStatus(state)} />;
 };
+export const mapAnalysisStateToLabel = (value: TaskState) => {
+  const presetKey: IconedStatusPreset = getTaskStatus(value);
+  const { t } = useTranslation();
+  const presets = buildPresets(t);
+  const label = presets[presetKey]?.label ?? t("terms.unknown");
+  return label;
+};
+export const buildPresets = (
```

Comment:
1. The purpose of the refactoring is to improve the code but keep the functionality the same.Here you change some of the labels - compared with IconedStatus component.
2. After we extracted the code to a separate reusable function we need to use it in the original place i.e.

```
 export const IconedStatus: React.FC<IIconedStatusProps> = ({
   preset,
   status,
@@ -62,11 +83,12 @@ export const IconedStatus: React.FC<IIconedStatusProps> = ({
   tooltipCount = 0,
 }: IIconedStatusProps) => {
   const { t } = useTranslation();
+  const messages = buildPresetsLabels(t);
   const presets: IconedStatusPresetType = {
     InProgressInheritedReviews: {
       icon: <InProgressIcon />,
       status: "info",
-      label: t("terms.inProgress"),
+      label: messages.InProgressInheritedReviews.label,
       tooltipMessage: t("message.inheritedReviewTooltip", {
         count: tooltipCount,
       }),
```

Reply:
Hi @rszwajko
Looks like `buildPresets` is not related only to analysis status but to any status (assessment, review, etc.) where do you suggest to move this functionality please?

Reply:
hmm...right. Probably better to put there in the same place as the original code - in the `client/src/app/components/Icons/IconedStatus.tsx`